src/tasks/batch_integration/metrics/isolated_label_asw/script.py

```diff
@@ -13,16 +13,15 @@
 ## VIASH END
 
 print('Read input', flush=True)
-adata = ad.read_h5ad(par['input_integrated'])
+input_solution = ad.read_h5ad(par['input_solution'])
+input_integrated = ad.read_h5ad(par['input_integrated'])
+input_solution.X = input_solution.layers['normalized']
+input_solution.obsm["X_emb"] = input_integrated.obsm["X_emb"]
 
-print('preprocess data', flush=True)
-adata.X = adata.layers['normalized']
-adata_int = adata.copy()
-
-print('compute score')
+print('compute score', flush=True)
 
 score = isolated_labels_asw(
```

Comment:
Can a 'layer' argument be added to the `isolated_labels_asw` function? This would save us the trouble of having to copy the normalized data to `.X`.

Reply:
This metric does not require .X, unless you don't yet have an embedding. Rest is same as for isolated_label_f1, but w/o needing neighbors

Reply:
I see!

Reply:
Fixed!